### PR TITLE
fix(simulator UI) : fixed the zoom-slider that was broken after Bootstrap upgrade

### PR DESCRIPTION
**Pull request by salmoneatenbybear**
Fixed the simulator UI { zoom-slider } that was broken after #5135 

image

## Summary by CodeRabbit

- **New Features**
	- Improved accessibility with added `aria-label` attributes for zoom controls.
	- Enhanced dropdown menus in the navigation bar for better user interaction.

- **Bug Fixes**
	- Resolved issues with dropdown visibility and user permissions affecting UI elements.

- **Style**
	- Updated styling for zoom slider controls to improve visibility and consistency.
	- Adjusted spacing and layout of footer social icons for better responsiveness.
	- Enhanced layout of the footer component for improved organization.

### CODE REVIEW

**Comment by salmoneatenbybear** [comment]
@tachyons created a seprate pr for the zoom-slider fix, pls check when you get time... Thankyou !

**Comment by tachyons** [comment]
As per screenshot, it still different from what we have on production. https://circuitverse.org/simulator

**Comment by salmoneatenbybear** [comment]
> As per screenshot, it still different from what we have on production. https://circuitverse.org/simulator

Ok, i will add the spacing between the icons, and decrease the size of the slider, so its mirrored with that in prod.

**Comment by tachyons** [comment]
Also check the spacing on social links on home page

**Comment by tachyons** [comment]
And CI is still broken

**Comment by salmoneatenbybear** [comment]
> Also check the spacing on social links on home page

sure, will do that

**Comment by salmoneatenbybear** [comment]
> And CI is still broken

It appears the CI is failing because it's using Node.js 22.x, and the canvas package doesn't have pre-built binaries for that version yet. As a result, it tries to compile from source but fails due to missing system dependencies like pangocairo. This issue isn't caused by my changes. Switching the CI to use Node.js 20.x LTS should resolve the problem, as canvas provides pre-built binaries for that version.

**Comment by salmoneatenbybear** [comment]
> And CI is still broken

i thought its better to consult with you before doing anything related to the ci config

**Comment by tachyons** [comment]
Edit layout dialog in simulator is also broken

**Comment by salmoneatenbybear** [comment]
@tachyons

 
image



image

**Comment by salmoneatenbybear** [comment]
> Edit layout dialog in simulator is also broken

will get it fixed, right away.

File: app/component/social_links_component.html.erb [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <div class="d-flex flex-row footer-social-icons-row">
   <% social_links.each do |link| %>
-    <%= link_to link[:url], class: "bounce-out-on-hover me-3", target: "_blank", data: { tooltip: link[:name] }, aria: { label: link[:name] } do %>
+    <%= link_to link[:url], class: "bounce-out-on-hover me-1", target: "_blank", data: { tooltip: link[:name] }, aria: { label: link[:name] } do %>
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/component/social_links_component.html.erb:3]: @tachyons should i do this ?

**Comment by salmoneatenbybear** [comment]
Fixed the Edit Layout Box and checked that the functionality is working too. 

@tachyons pls let me know if any other things need polishing or changes, would be happy to make them. 

image

File: app/views/simulator/edit.html.erb [hunk 2]
```diff
@@ -487,123 +485,97 @@
...
+        <button class="zoomButton-left btn btn-lg no-select" id="layoutTitleLeft">
+          <svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30">
+            <g fill="none" stroke="#f1f9ff" stroke-width="2">
+              <rect width="30" height="30" rx="10" stroke="none" />
+              <rect x="1" y="1" width="28" height="28" rx="9" fill="none" />
+            </g>
+            <path d="M4.7,6.1,0,1.4,1.4,0,4.7,3.3,8,0,9.4,1.4Z" transform="translate(18.1 9.655) rotate(90)" fill="#fff" />
+          </svg>
+        </button>
+
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:555]: done

File: app/views/simulator/edit.html.erb [hunk 3]
```diff
@@ -487,123 +485,97 @@
...
+        <button class="zoomButton-down btn btn-lg no-select" id="layoutTitleDown">
+          <svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30">
+            <g fill="none" stroke="#f1f9ff" stroke-width="2">
+              <rect width="30" height="30" rx="10" stroke="none" />
+              <rect x="1" y="1" width="28" height="28" rx="9" fill="none" />
+            </g>
+            <path d="M4.7,6.1,0,1.4,1.4,0,4.7,3.3,8,0,9.4,1.4Z" transform="translate(10 11.655)" fill="#fff" />
+          </svg>
+        </button>
+
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:545]: done

File: app/views/simulator/edit.html.erb [hunk 4]
```diff
@@ -487,123 +485,97 @@
...
       </button>
     </div>
-    <div class="">
+
+    <div class="d-flex align-items-center justify-content-between mb-2">
       <span>Reset all nodes:</span>
-      <button
-        class="zoomButton btn-lg"
-        id="layoutResetNodes">
+      <button class="zoomButton btn btn-lg" id="layoutResetNodes">
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:517]: done

File: app/views/simulator/edit.html.erb [hunk 5]
```diff
@@ -487,123 +485,97 @@
...
-            <rect width="30" height="30" rx="10" stroke="none" />
-            <rect x="1" y="1" width="28" height="28" rx="9" fill="none" />
-          </g>
-          <path id="Path_36" data-name="Path 36" d="M4.7,6.1,0,1.4,1.4,0,4.7,3.3,8,0,9.4,1.4Z" transform="translate(18.1 9.655) rotate(90)" fill="#fff" />
-        </svg>
+    <div class="layout-title mb-2">
+      <span>Title</span>
+      <div class="layout--btn-group d-flex gap-2">
+        <!-- Title Control Buttons with SVGs -->
+        <button class="zoomButton-up btn btn-lg no-select" id="layoutTitleUp">
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:526]: done

File: app/views/simulator/edit.html.erb [hunk 6]
```diff
@@ -487,123 +485,97 @@
...
-          <g id="Rectangle_1072" data-name="Rectangle 1072" fill="none" stroke="#f1f9ff" stroke-width="2">
-            <rect width="30" height="30" rx="10" stroke="none" />
-            <rect x="1" y="1" width="28" height="28" rx="9" fill="none" />
-          </g>
-          <path id="Path_36" data-name="Path 36" d="M4.7,6.1,0,1.4,1.4,0,4.7,3.3,8,0,9.4,1.4Z" transform="translate(12 19.055) rotate(-90)" fill="#fff" />
-        </svg>
-      </button>
-    </div>
+        
+        <button class="zoomButton-down btn btn-lg no-select" id="layoutTitleDown">
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:536]: done

File: app/views/simulator/edit.html.erb [hunk 7]
```diff
@@ -487,123 +485,97 @@
...
+          <svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30">
+            <g fill="none" stroke="#f1f9ff" stroke-width="2">
+              <rect width="30" height="30" rx="10" stroke="none" />
+              <rect x="1" y="1" width="28" height="28" rx="9" fill="none" />
+            </g>
+            <path d="M4.7,6.1,0,1.4,1.4,0,4.7,3.3,8,0,9.4,1.4Z" transform="translate(10 11.655)" fill="#fff" />
+          </svg>
+        </button>
+        
+        <button class="zoomButton-left btn btn-lg no-select" id="layoutTitleLeft">
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:546]: done

File: app/views/simulator/edit.html.erb [hunk 8]
```diff
@@ -487,123 +485,97 @@
...
+          <svg xmlns="http://www.w3.org/2000/svg" width="30" height="30" viewBox="0 0 30 30">
+            <g fill="none" stroke="#f1f9ff" stroke-width="2">
+              <rect width="30" height="30" rx="10" stroke="none" />
+              <rect x="1" y="1" width="28" height="28" rx="9" fill="none" />
+            </g>
+            <path d="M4.7,6.1,0,1.4,1.4,0,4.7,3.3,8,0,9.4,1.4Z" transform="translate(18.1 9.655) rotate(90)" fill="#fff" />
+          </svg>
+        </button>
+        
+        <button class="zoomButton-right btn btn-lg no-select" id="layoutTitleRight">
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:556]: done

File: app/views/simulator/edit.html.erb [hunk 9]
```diff
@@ -487,123 +485,97 @@
...
-    <div class="layout-title--enable">
+
+    <div class="layout-title--enable d-flex align-items-center justify-content-between mb-2">
       <span>Title Enabled:</span>
       <label class="switch">
-        <input
-          type="checkbox"
-          checked
-          id="toggleLayoutTitle">
+        <input type="checkbox" checked id="toggleLayoutTitle">
```
[the remark this answers is not in the document]
Comment (salmoneatenbybear) [reply app/views/simulator/edit.html.erb:571]: done

**Comment by salmoneatenbybear** [comment]
> > And CI is still broken
> 
> It appears the CI is failing because it's using Node.js 22.x, and the canvas package doesn't have pre-built binaries for that version yet. As a result, it tries to compile from source but fails due to missing system dependencies like pangocairo. This issue isn't caused by my changes. Switching the CI to use Node.js 20.x LTS should resolve the problem, as canvas provides pre-built binaries for that version.

@tachyons I think it fails because of this

**Comment by salmoneatenbybear** [comment]
superseded by #5169

**Pull request closed without merging**